src/run.h

```diff
@@ -62,6 +65,11 @@ class Run {
     // aborts this run
     bool abort();
 
+    // store metadata into run
+    bool tag(const std::string& key, const std::string& value);
+    // Return the meta as JSON text
+    std::string getMetaDataJsonString();
```

Comment:
I don't think `Run` should know about JSON at all. Just do

`const std::map<std::string, std::string>& metadata() const { return metadata; }`

and do the conversion to JSON in the `Laminar` class with a local method there.